### DISCUSSION TITLE
How does optimize_restart options works?

### DISCUSSION THREAD

**Discussion by diegodinamarca** [mhm-ufz/mHM#90] [General]
Hello mHM team, in the reference files I see two optimize options: optimize and optimize_restart. I know that with the first one we can optimize the model with an objective function, but when I tried the second option none optimization occur and I get the same outputs. 

when the optimize options is .true. it gives "ERROR: cannot read states from restart file when optimizing" so I'm feeling kind of lost about how to use this option.  

We are modifying the restart files using custom FC and PWP maps so we were wondering if it was possible to calibrate the model using the restart files. Any help would be very much appreciated.

Regards
Diego

**Comment by Bikash0101** [comment]
Although, I'm also new to this model, I guess the optimize restart works only when you set the write restart and read restart to TRUE in the mhm.nml file.

**Comment by StephanThober** [comment]
Hi Diego,

the optimize_restart and restart options do different things. The restart option allows to restart from a given state (contained in the restart file), like soil moisture, saturated soil water content etc. The optimize_restart option allows to restart the optimization. This optinon exists because the optimization can take a long time especcially if the domain you are optimizing the model for is large. If you do the optimization on compute clusters, these often have run time limits for the jobs. For example, a compute job is only allowed to run for 2 days, which might not be enough to optimize the model. In this case, this optimize_restart option saves the state of the calibration (not the state of the model!) and the optimization can proceed from this. This option is, however, only available for the SCE method.

Best,
Stephan

**Comment by diegodinamarca** [comment]
Thank you Stephan for your answer I understand now how it works
